### PR TITLE
Rule Collection – Improving the Schema

### PR DESCRIPTION
**Pull request by isaaclombardssw**
Added descriptions, and auto-population for new rules fields (filename, date created and also guid).

🤖🤖🤖

This pull request includes several enhancements to the `Rules` collection in the `tina/collections/rules.ts` file, focusing on adding descriptions and improving the user interface for various fields.

Enhancements to the `Rules` collection:

* Added a `defaultItem` function to generate default values for new rules, including a GUID, creation date, and filename. (`tina/collections/rules.ts`)
* Added descriptions to multiple fields to provide better context and guidance for users:
  * `title` field: "This is used as the major heading for the rule." (`tina/collections/rules.ts`)
  * `uri` field: "The URI of the rule – this defines the slug and references." (`tina/collections/rules.ts`)
  * `authors` field: "The list of contributors for this rule." (`tina/collections/rules.ts`)
  * `related` field: "The URIs of rules that should be suggested based on the content of this rule." (`tina/collections/rules.ts`)
  * `created` field: "If you see this field, contact a dev immediately 😳 (should be a hidden field generated in the background)." (`tina/collections/rules.ts`)
  * `archivedreason` field: "If this rule has been archived, summarise why here." (`tina/collections/rules.ts`)
  * `guid` field: "If you see this field, contact a dev immediately 😳 (should be a hidden field generated in the background)." (`tina/collections/rules.ts`)
  * `seoDescription` field: "A summary of the page content, used for SEO purposes. This can be generated automatically with AI." (`tina/collections/rules.ts`)
  * `body` field: "Rule content. This is where the magic happens." (`tina/collections/rules.ts`)
* Improved the UI for the `authors` field by adding item properties to display the contributor's name with an icon. (`tina/collections/rules.ts`)
* Made the `created` and `guid` fields hidden in the UI to avoid user modification. (`tina/collections/rules.ts`)

🤖🤖🤖

Relates to #1464.

### CODE REVIEW

File: tina/collections/rules.ts [hunk 1]
```diff
@@ -1,36 +1,59 @@
...
 
 export const Rules: Collection = {
   label: 'Rules',
   name: 'rules',
   path: '/rules',
   format: 'md',
+  defaultItem: () => {
+    return {
+      guid: generateGuid(),
+      created: new Date().toISOString(),
```
Comment (bradystroud) [inline tina/collections/rules.ts:12]: check if we actually use this anywhere (I think no)

File: tina/collections/rules.ts [hunk 2]
```diff
@@ -51,27 +75,40 @@ export const Rules: Collection = {
...
       type: 'string',
       name: 'seoDescription',
       label: 'SEO Description',
+      description: 'A summary of the page content, used for SEO purposes. This can be generated automatically with AI.',
     },
     {
       type: 'rich-text',
       name: 'body',
       label: 'Body',
+      description: 'Rule content. This is where the magic happens.',
```
Comment (bradystroud) [inline tina/collections/rules.ts:111]: I would expect a link to the wiki (I know its not relevant at the moment)

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: tina/collections/rules.ts (hunk 2 above)
Comment (isaaclombardssw) [reply tina/collections/rules.ts:111]: The wiki? 🤔

File: tina/collections/rules.ts (hunk 1 above)
Comment (isaaclombardssw) [reply tina/collections/rules.ts:12]: The guid or the date?

Comment (bradystroud) [reply tina/collections/rules.ts:12]: created date sorry - i think we just use the actual git history now

File: tina/collections/rules.ts (hunk 2 above)
Comment (bradystroud) [reply tina/collections/rules.ts:111]: A new section here
https://github.com/SSWConsulting/SSW.Rules.Content/wiki/How-to-Edit-Rules#how-to-edit-rules-using-netlify-cms

Comment (isaaclombardssw) [reply tina/collections/rules.ts]: ✅ Done

File: tina/collections/rules.ts (hunk 1 above)
Comment (isaaclombardssw) [reply tina/collections/rules.ts:13]: I can't tell, but it should probably still exist in the schema to fit with the rule layout until we start changing the md formats. I've created a PBI and linked it https://github.com/SSWConsulting/SSW.Rules/issues/1543.

**Review by bradystroud — DISMISSED** [review]

**Review by bradystroud — APPROVED** [review]

**Pull request merged**